mass_mailing_custom_unsubscribe/static/src/js/unsubscribe.js

```diff
@@ -5,46 +5,46 @@
  * to extend it as it is currently designed. It is almost a copy+paste from
  * upstream, to allow easier version/patch updates, so linter is disabled. */
```

Comment:
As you see here, this is almost a copy of upstream (find it in https://github.com/odoo/odoo/blob/14.0/addons/mass_mailing/static/src/js/unsubscribe.js).

In these specific cases, I don't recommend modernizing the code too much. This way, we can keep it more similar to upstream, and it's easier to extract the diff to know exactly what has changed.

Also you can see that one line below, we disable ESLint (and that's the reason: to avoid linter failures because we want the file to be "badly formatted" for the sake of being closer to upstream code).

So, although I appreciate your effort on modernizing this file's code, I have to ask you to leave it as it was please. More specifically: copy the v14 version of this file, paste it, and keep the important part of the diff.

Another option you have is to try to do it without replacing the file, by overriding something. For example, in the qweb template you can replace any element ID where we need a different logic, so we only need to override that part, and the JS code that attaches events here doesn't trigger because the element with the old ID isn't found. Or to remove the events that upstream attached and replace it with yours.

Reply:
Thanks @Yajo , you're right, I shouldn't have modernize the file since it's almost a copy of Upstream. As you suggested, I copied the code from version 14.0 of the upstream file and kept the important parts of the diff and I added ``// prettier-ignore`` comment to avoid re-formating by Prettier. Now we can clearly see the modifications to the original code.